prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2638,3 +2638,71 @@ void rxr_ep_record_tx_op_completed(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_
 	}
 }
 
+#if EFA_UNIT_TEST
+
+#include "efa_unit_tests.h"
+
+static void check_ep_pkt_pool_flags(struct efa_resource resource, int expected_flags)
+{
+       struct fid_ep *ep;
+       struct rxr_ep *rxr_ep;
+       int ret;
+       ret = fi_endpoint(resource.domain, resource.info, &ep, NULL);
+       assert_int_equal(ret, 0);
+       rxr_ep = container_of(ep, struct rxr_ep, util_ep.ep_fid);
+       assert_int_equal(rxr_ep->efa_tx_pkt_pool->attr.flags, expected_flags);
+       assert_int_equal(rxr_ep->efa_rx_pkt_pool->attr.flags, expected_flags);
+       fi_close(&ep->fid);
+}
+
+/**
+ * @brief Test the pkt pool flags in rxr_ep_init()
+ */
+void test_rxr_ep_pkt_pool_flags()
+{
+	int ret;
+	struct efa_resource resource = {0};
+	ret = efa_resource_create(&resource);
+	assert_int_equal(ret, 0);
+
+	g_efa_fork_status = EFA_FORK_SUPPORT_ON;
+	check_ep_pkt_pool_flags(resource, OFI_BUFPOOL_NONSHARED);
+
+	g_efa_fork_status = EFA_FORK_SUPPORT_OFF;
+	check_ep_pkt_pool_flags(resource, OFI_BUFPOOL_HUGEPAGES);
+
+	g_efa_fork_status = EFA_FORK_SUPPORT_UNNEEDED;
+	check_ep_pkt_pool_flags(resource, OFI_BUFPOOL_HUGEPAGES);
+
+	efa_resource_destroy(&resource);
+}
+
+/**
+ * @brief When the buf pool is created with OFI_BUFPOOL_NONSHARED,
+ * test if the allocated memory is page aligned.
+ */
+void test_rxr_ep_pkt_pool_page_alignment()
```

Comment:
since this test does not use static funciton, may be put is in `efa_unit_tests.c` or `efa_unit_test_ep.c`?

Reply:
Done.

Reply:
Actually both functions in `rxr_ep.c` can be moved to `efa_unit_test_ep.c`. Updated in the newest revision.